custom_components/hyxi_cloud/coordinator.py

```
@@ -148,6 +162,23 @@ async def _async_update_data(self):
+                self._handle_update_error(err)
+                raise
+
+            cached_devices = await self.device_store.async_load()
+            if cached_devices:
+                self.hyxi_metadata["last_error"] = str(err)
+                self.hyxi_metadata["api_status"] = "Offline"
+                _LOGGER.error(
+                    "HYXI Cloud API fetch failed: %s. Falling back to %d cached devices from storage.",
+                    err,
+                    len(cached_devices),
+                )
```

Ruff (0.15.20)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)